src/components/Trade/hooks/useDonationAmountBelowMinimum.tsx

```diff
@@ -1,30 +1,41 @@
-import { thorchainAssetId } from '@shapeshiftoss/caip'
+import { ethAssetId, thorchainAssetId } from '@shapeshiftoss/caip'
 import { useMemo } from 'react'
 import { bnOrZero } from 'lib/bignumber/bignumber'
 import { SwapperName } from 'lib/swapper/api'
 import { RUNE_OUTBOUND_TRANSACTION_FEE_CRYPTO_HUMAN } from 'lib/swapper/swappers/ThorchainSwapper/constants'
-import { selectMarketDataById } from 'state/slices/marketDataSlice/selectors'
-import { useAppSelector } from 'state/store'
-import { selectDonationAmountFiat } from 'state/zustand/swapperStore/amountSelectors'
+import { selectFeeAssetByChainId } from 'state/slices/assetsSlice/selectors'
+import { selectFiatToUsdRate, selectMarketDataById } from 'state/slices/marketDataSlice/selectors'
+import { store, useAppSelector } from 'state/store'
+import { selectDonationAmountUsd } from 'state/zustand/swapperStore/amountSelectors'
+import { selectSellAsset } from 'state/zustand/swapperStore/selectors'
 import { useSwapperStore } from 'state/zustand/swapperStore/useSwapperStore'
 
 export const useDonationAmountBelowMinimum = () => {
-  const runePrice = useAppSelector(state => selectMarketDataById(state, thorchainAssetId)).price
-  const donationAmountFiat = useSwapperStore(selectDonationAmountFiat)
+  const runePriceUsd = useAppSelector(state => selectMarketDataById(state, thorchainAssetId)).price
+  const donationAmountUsd = useSwapperStore(selectDonationAmountUsd)
   const activeSwapperName = useSwapperStore(state => state.activeSwapperWithMetadata?.swapper.name)
+  const sellAsset = useSwapperStore(selectSellAsset)
+  const sellAssetChainId = sellAsset?.chainId
+  const feeAsset = useAppSelector(state => selectFeeAssetByChainId(state, sellAssetChainId ?? ''))
+  const buyAmountFiat = useSwapperStore(state => state.buyAmountFiat)
+  const selectedCurrencyToUsdRate = selectFiatToUsdRate(store.getState())
+  const buyAmountUsd = bnOrZero(buyAmountFiat).div(selectedCurrencyToUsdRate)
 
   // Some swappers have a minimum donation amount, whereby collecting below that amount will result in the donated amount being lost
   const isDonationAmountBelowMinimum = useMemo(() => {
     switch (activeSwapperName) {
       case SwapperName.Thorchain: {
-        return bnOrZero(donationAmountFiat)
-          .div(runePrice)
+        return bnOrZero(donationAmountUsd)
+          .div(runePriceUsd)
           .lte(RUNE_OUTBOUND_TRANSACTION_FEE_CRYPTO_HUMAN)
       }
+      case SwapperName.Zrx:
+      case SwapperName.OneInch:
+        return feeAsset?.assetId === ethAssetId && buyAmountUsd.lt(500)
```

Comment:
abstract to constant a la `RUNE_OUTBOUND_TRANSACTION_FEE_CRYPTO_HUMAN`?

Reply:
Good call. Renamed to `MINIMUM_DONATION_RECEIVE_AMOUNT_USD_FROM_ETH_CHAIN` (names are hard) in https://github.com/shapeshift/web/pull/4536/commits/eb436ae4d89f75be6a44846f21d59c0e1deb772c